### ISSUE TITLE
PermissionError raised when tab completing

### ISSUE THREAD

**Issue by asmeurer** [xonsh/xonsh#1033]
I pressed tab after the `rea`

```pytb
sympy-paperpre-print🐚 e reaException in thread Thread-32:
Traceback (most recent call last):
  File "/Users/aaronmeurer/anaconda3/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/Users/aaronmeurer/anaconda3/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/aaronmeurer/anaconda3/lib/python3.5/site-packages/prompt_toolkit/interface.py", line 759, in run
    completions = list(buffer.completer.get_completions(document, complete_event))
  File "/Users/aaronmeurer/anaconda3/lib/python3.5/site-packages/xonsh/ptk/completer.py", line 36, in get_completions
    self.ctx)
  File "/Users/aaronmeurer/anaconda3/lib/python3.5/site-packages/xonsh/completer.py", line 224, in complete
    if cmd in self._cmds_cache:
  File "/Users/aaronmeurer/anaconda3/lib/python3.5/site-packages/xonsh/tools.py", line 1142, in __contains__
    return item in self.all_commands
  File "/Users/aaronmeurer/anaconda3/lib/python3.5/site-packages/xonsh/tools.py", line 1174, in all_commands
    allcmds |= set(executables_in(path))
  File "/Users/aaronmeurer/anaconda3/lib/python3.5/site-packages/xonsh/tools.py", line 329, in executables_in
    yield from (x.name for x in scandir(path)
  File "/Users/aaronmeurer/anaconda3/lib/python3.5/site-packages/xonsh/tools.py", line 330, in <genexpr>
    if x.is_file() and os.access(x.path, os.X_OK))
PermissionError: [Errno 13] Permission denied: '/usr/sbin/weakpass_edit'
```

It should have completed `README.md`.

**Comment by gforsyth** [comment]
Thanks for reporting, @asmeurer -- what is `e` aliased to?

**Comment by melund** [comment]
I think this an other error in the new `executables_in()` function. See
https://github.com/scopatz/xonsh/pull/1030 ... We have a similar function
`yield_executables()` which shouldn't have this error since it checks for
permissions.
https://github.com/scopatz/xonsh/blob/d69c09f4708a9efe946ef780205ffb9ca87ee79b/xonsh/environ.py#L785

Maybe we could switch to `yield_executables()` as temp fix.

**Comment by melund** [comment]
Ok. I looked into it a little further. The functions were different. `yield_executables()` looks for executables with a certain name in a directory. While `executables_in()` lists all executables in a certain directory. I think the functions should still be merged as they duplicate functionality. But right now we better fix `executables_in()`

**Comment by asmeurer** [comment]
https://github.com/asmeurer/dotfiles/blob/ca4b8fdf07f22ac28e59ef6090304f028b169bed/.xonshrc#L189

**Comment by adqm** [comment]
It looks like this was fixed with #1035, so I'm going to close this issue.  If you see this problem again, please feel free to reopen.
